hummingbot/market/market_base.pyx

```diff
@@ -217,12 +227,34 @@ cdef class MarketBase(NetworkIterator):
         """
         return self._account_balances.get(currency, s_decimal_0)
 
+    def available_balance_limit_applied(self, currency: str, limit: Decimal) -> Decimal:
+        """
+        Apply budget limit on an available balance, the limit is calculated as followings:
+        - Minus balance used in outstanding orders (in flight orders), if the budget is 1 ETH and the bot has already
+          used 0.5 ETH to put a maker buy order, the budget is now 0.5
+        - Plus balance accredited from filled orders (since the bot started), if the budget is 1 ETH and the bot has
+          bought LINK (for 0.5 ETH), the ETH budget is now 0.5. However if later on the bot has sold LINK (for 0.5 ETH)
+          the budget is now 1 ETH
+        """
+        in_flight_balance = self.in_flight_asset_balances(self.in_flight_orders).get(currency.upper(), s_decimal_0)
+        limit -= in_flight_balance
+        filled_balance = self.order_filled_balances().get(currency.upper(), s_decimal_0)
+        limit += filled_balance
+        asset_limit = max(limit, s_decimal_0)
+        available_balance = self._account_available_balances.get(currency.upper(), s_decimal_0)
```

Comment:
I don't think this should be `.get(currency.upper(), s_decimal_0)` here since this is an exchange function and some use lower case names? It caused issues for me but changing to `.get(currency, s_decimal_0)` fixed it for me.

Reply:
You're right. Fixed now.

Reply:
> You're right. Fixed now.

👍